core/shared/src/main/scala/zio/Chunk.scala

```diff
@@ -1117,6 +1160,80 @@ object Chunk {
     override def toArray[A1 >: A](n: Int, dest: Array[A1]): Unit = { val _ = vector.copyToArray(dest, n, length) }
   }
 
+  private[zio] object Buff {
+    final def arrayBacked[A](buffer: Buffer): Option[Chunk[A]] =
+      if (!buffer.hasArray()) None
+      else {
+        val pos = buffer.position()
```

Comment:
Is there an advantage to copying the array manually rather than using `get`? It appears that this method is redundant if not.

Reply:
And we are allocating an Option... Yeah, I guess we could get rid of it.

Reply:
that option allocation was going to be my next remark ;)

Reply:
Finally do you think inlining these methods into the `from*Buffer` methods would make sense?

Reply:
Doing that as we speak.

Reply:
Love it, most of the code is gone.